### PR TITLE
Fixed #26785 -- Set BaseSpatialField.empty_strings_allowed to False.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/26785

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, test on oracle.

File: tests/gis_tests/geoapp/test_functions.py [hunk 1]
```diff
@@ -223,12 +223,9 @@ def test_intersection(self):
         geom = Point(5, 23, srid=4326)
         qs = Country.objects.annotate(inter=functions.Intersection('mpoly', geom))
         for c in qs:
-            if spatialite or mysql:
-                # When the intersection is empty, Spatialite and MySQL return None
+            if spatialite or mysql or oracle:
```
Comment (timgraham) [inline tests/gis_tests/geoapp/test_functions.py]: This could be a separate commit or ticket, but could a database feature flag replace the vendor checks?

**Comment by timgraham** [comment]
For the commit message, I usually try to describe the issue that's fixed rather than the implementation.

File: tests/gis_tests/geoapp/test_functions.py (hunk 1 above)
Comment (sir-sigurd) [inline tests/gis_tests/geoapp/test_functions.py]: Yes, I thought about this, but I'm planning to do some adjacent changes and I think that it would more clear after that what feature flag we need to add.

**Comment by timgraham** [comment]
merged in ea4665066b651e0e2730f2e2d85fcef9d7ac7966, thanks!
